File: app/src/main/java/android/shts/jp/nogifeed/fragments/AboutFragment.java
package android.shts.jp.nogifeed.fragments;

import android.content.Context;
import android.os.Bundle;
import android.shts.jp.nogifeed.R;
import android.shts.jp.nogifeed.activities.BaseActivity;
import android.shts.jp.nogifeed.adapters.BindableAdapter;
import android.shts.jp.nogifeed.utils.IntentUtils;
import android.shts.jp.nogifeed.utils.PicassoHelper;
import android.shts.jp.nogifeed.utils.TrackerUtils;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ImageView;
import android.widget.ListView;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

public class AboutFragment extends Fragment {

    private static final String TAG = AboutFragment.class.getSimpleName();
    private static final String URL_ICON = "https://avatars1.githubusercontent.com/u/7928836?v=3&s=460";
    private ListView mAboutList;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    private void setupAboutListAdapter() {
        List<AboutItem> abouts = new ArrayList<AboutItem>();
        abouts.add(new AboutItem(getResources().getString(R.string.about_item_share),
                R.drawable.ic_social_share, new OnClickListener() {
            @Override
            public void onClick() {
                IntentUtils.recomendApp(getActivity());
            }
        }));
        abouts.add(new AboutItem(getResources().getString(R.string.about_item_rate),
                R.drawable.ic_action_thumb_up, new OnClickListener() {
            @Override
            public void onClick() {
                IntentUtils.rateApp(getActivity());
            }
        }));
        abouts.add(new AboutItem(getResources().getString(R.string.about_item_mention),
                R.drawable.ic_communication_messenger, new OnClickListener() {
            @Override
            public void onClick() {
                IntentUtils.inquiryApp(getActivity());
            }
        }));
        mAboutList.setAdapter(new AboutListAdapter(getActivity(), abouts));
    }

    private void setupListHeader(LayoutInflater inflater, ListView listView) {
        View view = inflater.inflate(R.layout.about_header, null);

        ImageView developerIcon = (ImageView) view.findViewById(R.id.developer_icon);
        developerIcon.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                IntentUtils.showDeveloper(getActivity());
                TrackerUtils.sendTrack(getActivity(), TAG, "OnClicked", "developerIcon");
            }
        });
        PicassoHelper.loadAndCircleTransform(getActivity(), developerIcon, URL_ICON);

        listView.addHeaderView(view);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_about, null);
        mAboutList = (ListView) view.findViewById(R.id.about_list);
        mAboutList.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
                ListView listView = (ListView) adapterView;
                AboutItem aboutItem = (AboutItem) listView.getItemAtPosition(i);

                if (aboutItem != null && aboutItem.listener != null) {
                    aboutItem.listener.onClick();
                    TrackerUtils.sendTrack(getActivity(), TAG, "OnClicked", aboutItem.title);
                }
            }
        });
        setupListHeader(inflater, mAboutList);
        setupAboutListAdapter();
        return view;
    }

    class AboutItem {
        public final String title;
        public final int iconRes;
        public final OnClickListener listener;
        AboutItem(String title, int iconRes, OnClickListener listener) {
            this.title = title;
            this.iconRes = iconRes;
            this.listener = listener;
        }
    }
    public interface OnClickListener {
        public void onClick();
    }

    static class AboutListAdapter extends BindableAdapter {

        class ViewHolder {
            TextView textView;
            ImageView imageView;
            ViewHolder(View view) {
                textView = (TextView) view.findViewById(R.id.title);
                imageView = (ImageView) view.findViewById(R.id.icon);
            }
        }

        public AboutListAdapter(Context context, List<AboutItem> list) {
            super(context, list);
        }

        @Override
        public View newView(LayoutInflater inflater, int position, ViewGroup container) {
            View view = inflater.inflate(R.layout.list_item_about_action, null);
            final ViewHolder holder = new ViewHolder(view);
            view.setTag(holder);
            return view;
        }

        @Override
        public void bindView(Object item, int position, View view) {
            final ViewHolder holder = (ViewHolder) view.getTag();
            AboutItem aboutLocalItem = (AboutItem) item;
            holder.imageView.setImageResource(aboutLocalItem.iconRes);
            holder.textView.setText(aboutLocalItem.title);
        }
    }
}

File: app/src/main/java/shts/jp/android/nogifeed/fragments/AboutFragment.kt
package shts.jp.android.nogifeed.fragments

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.about_header.view.*
import kotlinx.android.synthetic.main.fragment_about.*
import kotlinx.android.synthetic.main.list_item_about_action.view.*
import shts.jp.android.nogifeed.R
import shts.jp.android.nogifeed.utils.IntentUtils
import shts.jp.android.nogifeed.utils.PicassoHelper

class AboutFragment : Fragment() {

    companion object {
        private val URL_ICON = "https://avatars1.githubusercontent.com/u/7928836?v=3&s=460"

        fun newInstance(): AboutFragment = AboutFragment()
    }

    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_about, null)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        toolbar.setTitle(R.string.nav_menu_about_app)
        toolbar.setNavigationIcon(R.drawable.ic_clear_white_18dp)
        toolbar.setNavigationOnClickListener { activity.finish() }

        aboutHeaderView.developerIconView.apply {
            PicassoHelper.loadAndCircleTransform(activity, this, URL_ICON)
            setOnClickListener { IntentUtils.showDeveloper(context) }
        }

        socialShareListItem.apply {
            setOnClickListener { IntentUtils.recommendApp(context) }
            iconView.setImageResource(R.drawable.ic_action_share)
            titleTextView.setText(R.string.about_item_share)
        }

        thumbUpListItem.apply {
            setOnClickListener { IntentUtils.rateApp(context) }
            iconView.setImageResource(R.drawable.ic_action_thumb_up)
            titleTextView.setText(R.string.about_item_rate)
        }

        mentionListItem.apply {
            setOnClickListener { IntentUtils.inquiryApp(context) }
            iconView.setImageResource(R.drawable.ic_action_mode_comment)
            titleTextView.setText(R.string.about_item_mention)
        }
    }
}
